Change get_started to recommened Gz version per platform for installations

j-rivero opened the pull request:
# 🎉 New feature

## Summary

The PR tries to simplify the election of the Gz version for the users. Basically it provides a recommendation in the landing page of main docs page https://gazebosim.org/docs and the landing page of releases (i.e https://gazebosim.org/docs/garden). It also includes a clear warning about Windows and leave it out of the table.

The PR will require some effort when we release a new version or when a version is EOL but I think that could worth it.

### Current situation problems

I think that there is a number of issues currently that are a source of confusion:

 * The Gz version selected in the top left corner named "Release" is not preserved when clicking links. 
 * The Gz version selected by default in the top left corner named "Release" is the one in development which leaves the user without binary install methods.

Both points in combination cause the effect of:
When being in a landing page of a given distribution (i.e Garden https://gazebosim.org/docs/garden), there are two links in the first step 1 install point named `latest installation tutorial here` and `Please visit the main documentation pages for a list of all releases` both are redirecting to Harmonic (development version).

### Follow up work

A new PR to be sent to clarify the options for the ROS users in the same way that [we have for gazebo-classic](https://classic.gazebosim.org/tutorials?tut=ros_wrapper_versions&cat=connect_ros).

## Checklist
- [x] Signed all commits for DCO

mabelzhang commented on get_started.md:
```diff
@@ -7,10 +7,28 @@ simulation using Gazebo.
+| Mac Catalina | [Gz Garden](docs/garden/install_osx) (recommended), [Gz Fortress](docs/fortress/install_osx) and [Gz Citadel](docs/citadel/install_osx)
+
+Windows support via conda-forge is not fully functional, there are known runtime problems
+[stoping Gz to work](https://github.com/gazebosim/gz-sim/issues/168).
```
```suggestion
[stopping Gz to work](https://github.com/gazebosim/gz-sim/issues/168).
```

mabelzhang commented on get_started.md:
```diff
@@ -7,10 +7,28 @@ simulation using Gazebo.
+Windows support via conda-forge is not fully functional, there are known runtime problems
+[stoping Gz to work](https://github.com/gazebosim/gz-sim/issues/168).
+
+If the desired plaform is not listed above or if a particular feature in a
+given Gz release is needed, there is a installation package per release
```
```suggestion
If the desired platform is not listed above or if a particular feature in a
given Gz release is needed, there is an installation package per release
```

mabelzhang commented on get_started.md:
```diff
@@ -7,10 +7,28 @@ simulation using Gazebo.
+|Platform|Gz Versions|
+|---|---|
+| Ubuntu 22.04 Jammy | [Gz Garden](docs/garden/install_ubuntu) (recommended) and [Gz Fortress](docs/fortress/install_ubuntu)
+| Ubuntu 20.04 Focal | [Gz Garden](docs/garden/install_buuntu) (recommended), [Gz Fortress](docs/fortress/install_ubuntu) and [Gz Citadel](docs/citadel/install_ubuntu)
```
```suggestion
| Ubuntu 20.04 Focal | [Gz Garden](docs/garden/install_ubuntu) (recommended), [Gz Fortress](docs/fortress/install_ubuntu) and [Gz Citadel](docs/citadel/install_ubuntu)
```

mabelzhang reviewed (commented):
Cosmetic corrections.

You know the corresponding versions better than me, so I'll trust you on that :sweat_smile: Unless someone else wants to check that.

j-rivero commented:
> Cosmetic corrections.

Thanks so much!

azeey commented:
> The Gz version selected by default in the top left corner named "Release" is the one in development which leaves the user without binary install methods.

@nkoenig is looking into making the current stable version the default. If that's not visible, we can remove `Harmonic` from https://github.com/gazebosim/docs/blob/master/index.yaml until we open the tutorial party.

kscottz commented on get_started.md:
```diff
@@ -7,10 +7,28 @@ simulation using Gazebo.
+| Mac Catalina | [Gz Garden](docs/garden/install_osx) (recommended), [Gz Fortress](docs/fortress/install_osx) and [Gz Citadel](docs/citadel/install_osx)
+
+Windows support via conda-forge is not fully functional, there are known runtime problems
+[stopping Gz to work](https://github.com/gazebosim/gz-sim/issues/168).
```
```suggestion
[Gz for Windows Issue](https://github.com/gazebosim/gz-sim/issues/168).
```

kscottz commented on get_started.md:
```diff
@@ -7,10 +7,28 @@ simulation using Gazebo.
-read [the latest installation tutorial here](/docs/latest/install). Please
-visit the [main documentation](/docs) pages for a list of all releases,
-along with links to their respective installation tutorials.
+The recommended installation for non expert users is the use of binary
```
```suggestion
The recommended installation for new users is the use of binary
```

kscottz commented on get_started.md:
```diff
@@ -7,10 +7,28 @@ simulation using Gazebo.
+Windows support via conda-forge is not fully functional, there are known runtime problems
+[stopping Gz to work](https://github.com/gazebosim/gz-sim/issues/168).
+
+If the desired platform is not listed above or if a particular feature in a
```
```suggestion
If the desired platform is not listed above, or if a particular feature in a
```

kscottz reviewed (approved):
Better than what we have by a long shot! I approve given a few small tweaks.

azeey reviewed (approved):
Looks great! My only comment is that we replace all instances of `Gz` with `Gazebo`. I think adding another name (Gz) would only add to the confusion.

j-rivero commented:
> Looks great! My only comment is that we replace all instances of `Gz` with `Gazebo`. I think adding another name (Gz) would only add to the confusion.

Good point. Changed in [851eac8](https://github.com/gazebosim/docs/pull/338/commits/851eac8969b6d386263f976e2abfe5c5e58a1641)

I missed a quick reference to the EOL dates for the distributions and added that in [a7b9883](https://github.com/gazebosim/docs/pull/338/commits/a7b9883ae0c2f67782604f2d6cbd95f1611ab982). Sorry for the inconvenience, @kscottz @mabelzhang @azeey if you can give it a quick look at that commit. Would be great.

azeey commented:
@j-rivero if we can merge this in the next hour, I'll go ahead and deploy it with #341 .
